viewing_party/party.py

```diff
@@ -0,0 +1,168 @@
+    return most_watched_value
+
+def get_unique_watched(user_data):
+    seen = []
+    show_friends = []
+    for i in range(len(user_data["friends"])):
+        seen += user_data["friends"][i]["watched"]
+    for i in range(len(user_data["watched"])):
+        if user_data["watched"][i] not in seen:
+            show_friends.append(user_data["watched"][i])
+
+    if len(show_friends) == 0:
+        return []
+    return show_friends
+
+def get_friends_unique_watched(user_data):
+    seen = []
+    friends_show = []
+    for i in range(len(user_data["friends"])):
+        seen += user_data["friends"][i]["watched"]
+    no_repeats  = []
+    for i in friends_show:
+        if i not in no_repeats:
+            no_repeats.append(i)
+    friends_show = no_repeats
+    for each in seen:
+        if each not in user_data["watched"]:
+            friends_show.append(each)
+
+    if len(friends_show) == 0:
+        return []
+    no_repeats = []
+    for i in friends_show:
+        if i not in no_repeats:
+            no_repeats.append(i)
+    friends_show = no_repeats
+    return friends_show
+
+def get_available_recs(user_data):
+    seen = []
+    rec_list = []
+    for i in range(len(user_data["friends"])):
+        seen += user_data["friends"][i]["watched"]
+    user_seen = []
+    for element in user_data["watched"]:
+        user_seen.append(element["title"])
+    for each in seen:
+        if each["title"] not in user_seen:
+            rec_list.append(each)
+    print (rec_list)
+
+    subbed_only= []
+    for dict in rec_list:
+        if dict["host"] in user_data["subscriptions"]:
+            subbed_only.append(dict)
+    rec_list = subbed_only
+    if len(rec_list) == 0:
+        return []
+    no_repeats = []
+    for i in rec_list:
+        if i not in no_repeats:
+            no_repeats.append(i)
+    rec_list = no_repeats
+    return rec_list
+
+def get_new_rec_by_genre(user_data):
+    genre_counter = {}
+    rec_list = []
+    if user_data["watched"] == []:
+        return []
+    for list in user_data["watched"]:
+        if list["genre"] not in genre_counter.keys():
+            genre_counter[list["genre"]] = 1
+        else:
+            genre_counter[list["genre"]] += 1
+    most_genred = max(genre_counter, key = lambda x : x[1])
+    for list in user_data["friends"]:
+        for sublist in list["watched"]:
+            if sublist["genre"] in most_genred:
+                
+                rec_list.append(sublist)
+        no_repeats = []
+        for i in rec_list:
+            if i not in no_repeats:
+                no_repeats.append(i)
+        rec_list = no_repeats
+
+    return rec_list
+    
+def get_rec_from_favorites(user_data):
+    seen = []
+    show_friends = []
+    for i in range(len(user_data["friends"])):
+        seen += user_data["friends"][i]["watched"]
+    for i in range(len(user_data["watched"])):
+        if user_data["watched"][i] not in seen:
+            show_friends.append(user_data["watched"][i])
```

Comment:
Similar logic is used for `get_friends_unique_watched()`. You could definitely reuse that function here.